rust/arrow/src/compute/kernels/sort.rs

```diff
@@ -517,20 +650,32 @@ where
             },
         );
 
-    if !options.descending {
-        valids.sort_by(|a, b| cmp_array(a.1.as_ref(), b.1.as_ref()))
+    let mut len = values.len();
+    let descending = options.descending;
+
+    if let Some(size) = limit {
+        len = size.min(len);
+    }
+
+    // we are not using partial_sort here, because array is ArrayRef. Something is not working good in that.
```

Comment:
What does "not working good" mean here?

Reply:
Sorry, there is a bug in the previous version. It's fixed in the newest [commit](https://github.com/sundy-li/partial_sort/commit/372778d0011b4d647ed4ee13d1dacb0865eb266d#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759R103-R141)